consensus/enclave/impl/src/lib.rs

```diff
@@ -697,20 +708,79 @@ impl ConsensusEnclave for SgxConsensusEnclave {
             &fee_public_key.view_public_key,
         );
 
-        let fee_outputs = total_fees_by_token_id
+        // Compute the total fees for each known token id.
+        let mut total_fees: CtTokenMap<u128> = ct_min_fee_map.keys().cloned().collect();
+        for tx in transactions.iter() {
+            let token_id = TokenId::from(tx.prefix.token_id);
+            total_fees.add(&token_id, tx.prefix.fee as u128);
+        }
+
+        // Sort the token ids which did not appear to the end.
+        total_fees.sort_zeroes_to_end();
```

Comment:
To double tap - an attacker would be unaware that the fees outputs sorted are indeed zeroes due to CT sorting right?

Reply:
they would be unaware which token ids had zero fees and nonzero fees, which is the goal because if i know which token id has nonzero fees then I know which one you used in your transaction

Reply:
Thanks for the distinction